pyincore/analyses/commercialbuildingrecovery/commercialbuildingrecovery.py

```diff
@@ -0,0 +1,418 @@
+        delay_time.insert(0, 'guid', commercial_damage.reset_index(drop=True)['guid'])
+
+        # Perform an inner join to ensure only buildings with damage states are processed
+        merged_delay = pd.merge(sample_damage_states, delay_time, on='guid')
+
+        # Obtain the guids
+        merged_delay_guids = merged_delay['guid']
+
+        # Obtain the damage states
+        merged_delay_damage_states = merged_delay['sample_damage_states']
+
+        # Convert to numpy
+        samples_np = merged_delay.drop(columns=['guid', 'sample_damage_states']).to_numpy()
+
+        # First, we decompose redi_delay_factors into two dictionaries that can be used to compute vector operations
+        redi_idx = dict(zip(redi_delay_factors['Building_specific_conditions'], redi_delay_factors.index))
+
+        # Next, we produce two intermediate numpy matrices: one for med and one for sdv
+        redi_med = redi_delay_factors[['Ins_med', 'Enmo_med', 'Como_med', 'Per_med', 'Fin_med']].to_numpy()
+        redi_sdv = redi_delay_factors[['Ins_sdv', 'Enmo_sdv', 'Como_sdv', 'Per_sdv', 'Fin_sdv']].to_numpy()
+
+        # Define indices to facilitate interpretation of the code
+        inspection_idx = 0
+        engineer_idx = 1
+        contractor_idx = 2
+        permit_idx = 3
+        financing_idx = 4
+
+        for i, ds_list in enumerate(list(merged_delay_damage_states)):
+            samples_mcs = ds_list.split(",")
+
+            for j in range(num_samples):
+                # TODO: ask why there are many more damage states than samples
+                # Obtain the index for the corresponding damage state
+                dmg_state_idx = redi_idx[samples_mcs[j]]
+
+                # Use the index to select the appropriate mean and stdev vectors
+                mean_vec = redi_med[dmg_state_idx, :]
+                sdv_vec = redi_sdv[dmg_state_idx, :]
+
+                # Compute the delay vector
+                delay_vec = np.random.lognormal(np.log(mean_vec), sdv_vec)
+
+                if dmg_state_idx == 3 or dmg_state_idx == 2:
+                    delay_vec[4] = np.random.normal(mean_vec[4], sdv_vec[4])
+
+                # Compute the delay using that vector, already computed in the prior step
+                samples_np[i, j] = np.round(delay_vec[inspection_idx] +
+                                            np.max([
+                                                delay_vec[engineer_idx],
+                                                delay_vec[financing_idx],
+                                                delay_vec[contractor_idx]
+                                            ]) +
+                                            delay_vec[permit_idx])
+
+        total_delay = pd.DataFrame(samples_np, columns=colnames)
+        total_delay.insert(0, 'guid', merged_delay_guids)
+
+        return total_delay
+
+    def recovery_rate(self, buildings, sample_damage_states, total_delay):
```

Comment:
At first glance, this looks identical to the function in `pyincore/analyses/residentialbuildingrecovery/residentialbuildingrecovery.py.` You might be able to just reuse that